### PR TITLE
fix(aws): review checks with wrong attributes

### LINKED ISSUE #5498: Error when sending findings to SecurityHub: data.Resources[0].Id should NOT be shorter than 1 characters
### Steps to Reproduce

 Run the following command prowler aws --log-level ERROR -R arn:"${ACCOUNT_ROLE_ARN}" --security-hub --send-sh-only-fails -f "${REGION}" > /dev/null 2>&1 &  # Suppress output and run in the background

This command is run as part of running Prowler as an ECS task https://github.com/prowler-cloud/prowler/tree/master/contrib/aws/multi-account-securityhub

### Expected behavior

Findings should be sent to AWS Security Hub.

### Actual Result with Screenshots or Logs

Screenshot 2024-10-22 at 06 52 36

### How did you install Prowler?

Docker (docker pull toniblyx/prowler)

### Environment Resource

ECS Fargate

### OS used

Running on Docker image (ECR).

### Prowler version

4.5.0

### Pip version

N/A

### Context

_No response_

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context

Some checks did not have ARN, Resource Type or `Recommendation.Text`, which was making SecurityHub failing when sending findings.

Fix #5498.

### Description

- Review checks to complete wrong or empty values.
- Use the Severity Enum values when changing it within the checks.
- Improve WAF and Organizations service.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/ecr/ecr_registry_scan_images_on_push_enabled/ecr_registry_scan_images_on_push_enabled.py [hunk 1]
```diff
@@ -11,8 +11,7 @@ def execute(self):
                 report = Check_Report_AWS(self.metadata())
                 report.region = registry.region
                 report.resource_id = registry.id
-                # A registry cannot have tags
-                report.resource_tags = []
+                report.resource_arn = ecr_client.audited_account_arn
```
Comment (jfagoagas) [inline prowler/providers/aws/services/ecr/ecr_registry_scan_images_on_push_enabled/ecr_registry_scan_images_on_push_enabled.py]: I think here we should create our own ARN, what do you think?

File: prowler/providers/aws/services/rds/rds_instance_certificate_expiration/rds_instance_certificate_expiration.py [hunk 2]
```diff
@@ -67,11 +67,11 @@ def execute(self):
                         utc
                     ):
                         report.status = "FAIL"
-                        report.check_metadata.Severity = "high"
+                        report.check_metadata.Severity = Severity.high
```
Comment (jfagoagas) [inline prowler/providers/aws/services/rds/rds_instance_certificate_expiration/rds_instance_certificate_expiration.py:70]: Please, add a test to cover this scenario.

File: prowler/providers/aws/services/route53/route53_domains_privacy_protection_enabled/route53_domains_privacy_protection_enabled.py [hunk 3]
```diff
@@ -11,6 +11,7 @@ def execute(self) -> Check_Report_AWS:
         for domain in route53domains_client.domains.values():
             report = Check_Report_AWS(self.metadata())
             report.resource_id = domain.name
+            report.resource_arn = route53domains_client.audited_account_arn
```
Comment (jfagoagas) [inline prowler/providers/aws/services/route53/route53_domains_privacy_protection_enabled/route53_domains_privacy_protection_enabled.py]: The same again to create an ARN for this one.

File: prowler/providers/aws/services/route53/route53_domains_transferlock_enabled/route53_domains_transferlock_enabled.py [hunk 4]
```diff
@@ -11,6 +11,7 @@ def execute(self) -> Check_Report_AWS:
         for domain in route53domains_client.domains.values():
             report = Check_Report_AWS(self.metadata())
             report.resource_id = domain.name
+            report.resource_arn = route53domains_client.audited_account_arn
```
Comment (jfagoagas) [inline prowler/providers/aws/services/route53/route53_domains_transferlock_enabled/route53_domains_transferlock_enabled.py]: The same again to create an ARN for this one.

File: prowler/providers/aws/services/waf/waf_service.py [hunk 5]
```diff
@@ -14,15 +14,21 @@ def __init__(self, provider):
...
-        self._list_rules()
-        self.__threading_call__(self._get_rule, self.rules.values())
-        self._list_rule_groups()
-        self.__threading_call__(
-            self._list_activated_rules_in_rule_group, self.rule_groups.values()
-        )
-        self._list_web_acls()
-        self.__threading_call__(self._get_web_acl, self.web_acls.values())
-        self.__threading_call__(self._get_logging_configuration, self.web_acls.values())
+        if self.audited_partition == "aws":
```
Comment (jfagoagas) [inline prowler/providers/aws/services/waf/waf_service.py:17]: Why only `aws`? China has also CloudFront, at least by our regions file.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
That's a great one, so many improvements in our checks 🚀 Please review my comments when you get a chance, thanks!

File: prowler/providers/aws/services/waf/waf_service.py (hunk 5 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/waf/waf_service.py:17]: You can see in https://docs.aws.amazon.com/general/latest/gr/waf-classic.html that only this partition is supported.

**Comment by MrCloudSec** [comment]
@jfagoagas comments were solved!

**Review by jfagoagas — APPROVED** [review]
Thank you for addressing everything 🥇

File: prowler/providers/aws/services/route53/route53_domains_transferlock_enabled/route53_domains_transferlock_enabled.py (hunk 4 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/route53/route53_domains_transferlock_enabled/route53_domains_transferlock_enabled.py]: Talked IRL and decided to create a custom ARN.

File: prowler/providers/aws/services/route53/route53_domains_privacy_protection_enabled/route53_domains_privacy_protection_enabled.py (hunk 3 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/route53/route53_domains_privacy_protection_enabled/route53_domains_privacy_protection_enabled.py]: Talked IRL and decided to create a custom ARN.

File: prowler/providers/aws/services/ecr/ecr_registry_scan_images_on_push_enabled/ecr_registry_scan_images_on_push_enabled.py (hunk 1 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/ecr/ecr_registry_scan_images_on_push_enabled/ecr_registry_scan_images_on_push_enabled.py]: Talked IRL and decided to create a custom ARN.

**Pull request merged**